### ISSUE TITLE
Update Python Client Library to match our standardized API (adding methods as needed)

### ISSUE THREAD

**Issue by rufuspollock** [datopian/ckan3-py-sdk#17]
Pattern is in https://github.com/datopian/tech.datopian.com/issues/23

For full details on each method see https://github.com/datopian/ckan-client-js/issues/4 and associated sublinks.

## Acceptance

* [x] `Client` class with documented constructor **#18**

### Plumbing

* [x] `client.action(...)` for this just use existing library https://github.com/ckan/ckanapi **#18**

### Porcelain

NB: you can just implement on Dataset for metadataOnly=True for now.

Dataset

* [x] `create(datasetMetadata, metadataOnly=false)`: iterate through the resources pushing the data for each one then push the dataset metadata **#18**
* [x] `push(datasetMetadata, metadataOnly)`: ditto but for existing dataset **#18**
* [x] `retrieve(datasetIdOrName)` **#18**

Resource

* [x] `pushBlob` **#18**
* [ ] `createResource(resourceObject, datasetIdOrName, metadataOnly)`
* [ ] `pushResource(resourceObject, metadataOnly)`

Also provide example of creating a dataset from a single file ... ACTUALLY THIS SHOULD NOT HAPPEN. Always explicitly create the dataset.

If you just want to push / create metadata you can use the action API

use the mapper https://github.com/frictionlessdata/frictionless-ckan-mapper
```
import xxx as f11s2ckan_mapper
datasetFritionless = xxx
ckanDataset = f11s2ckan_mapper.dataset(datasetFriictionles)
client.action('package_create', ckan_metadata)
```

**Comment by cuducos** [comment]
There are only two methods not implemented in `master` ATM: `create_resource` and `push_resource` — should we create a new issue for them and close this issue, or is it better to leave this issue open that way?

**Comment by cuducos** [comment]
It was done before #18, and, after #18, fixed in e63d759eae8518749a9e2c2c8eb05a5ed814c335.

**Comment by mbeilin** [comment]
> There are only two methods not implemented in `master` ATM: `create_resource` and `push_resource` — should we create a new issue for them and close this issue, or is it better to leave this issue open that way?

@cuducos any updates on this? ^^
This is a blocker in NHS atm, cause the client needs a reproducible code example (python script) for:
1) create a dataset and add a resource (CSV) using the API.
2) update/add a new resource to an existing dataset using the API.

cc/ @rufuspollock
